lib/taskana-core/src/test/java/acceptance/classification/CreateClassificationAccTest.java

```diff
@@ -3,295 +3,319 @@
-    classification2.setServiceLevel("abc");
-    assertThatThrownBy(() -> CLASSIFICATION_SERVICE.createClassification(classification2))
-        .isInstanceOf(MalformedServiceLevelException.class);
+    return DynamicTest.stream(iterator, c -> String.format("for '%s'", c), test);
   }
 
   @WithAccessId(user = "businessadmin")
   @Test
-  void testCreateClassificationAlreadyExisting() throws Exception {
-    Classification classification = CLASSIFICATION_SERVICE.newClassification("Key3", "", "TASK");
-    classification.setServiceLevel("P1D");
-    Classification classificationCreated =
-        CLASSIFICATION_SERVICE.createClassification(classification);
-    assertThatThrownBy(() -> CLASSIFICATION_SERVICE.createClassification(classificationCreated))
-        .isInstanceOf(ClassificationAlreadyExistException.class);
+  void should_ThrowException_When_TryingToCreateClassificationWithInvalidKey() {
+    Classification classificationWithNullKey =
+        classificationService.newClassification(null, "DOMAIN_A", "TASK");
+
+    assertThatThrownBy(() -> classificationService.createClassification(classificationWithNullKey))
+        .isInstanceOf(InvalidArgumentException.class)
+        .hasMessage("Classification must contain a key");
   }
 
   @WithAccessId(user = "businessadmin")
   @Test
-  void testCreateClassificationInUnknownDomain() {
+  void should_ThrowException_When_TryingToCreateClassificationWithInvalidDomain() {
     Classification classification =
-        CLASSIFICATION_SERVICE.newClassification("Key3", "UNKNOWN_DOMAIN", "TASK");
-    assertThatThrownBy(() -> CLASSIFICATION_SERVICE.createClassification(classification))
-        .isInstanceOf(DomainNotFoundException.class);
+        classificationService.newClassification("KeyErrCreation", "UNKNOWN_DOMAIN", "TASK");
+    DomainNotFoundException expectedException = new DomainNotFoundException("UNKNOWN_DOMAIN");
+
+    assertThatThrownBy(() -> classificationService.createClassification(classification))
+        .isInstanceOf(DomainNotFoundException.class)
+        .usingRecursiveComparison()
+        .isEqualTo(expectedException);
   }
 
   @WithAccessId(user = "businessadmin")
   @Test
-  void testCreateClassificationOfUnknownType() {
+  void should_ThrowException_TryingToCreateClassificationWithInvalidType() {
     Classification classification =
-        CLASSIFICATION_SERVICE.newClassification("Key3", "DOMAIN_A", "UNKNOWN_TYPE");
-    assertThatThrownBy(() -> CLASSIFICATION_SERVICE.createClassification(classification))
-        .isInstanceOf(InvalidArgumentException.class);
+        classificationService.newClassification("KeyErrCreation", "DOMAIN_A", "UNKNOWN_TYPE");
+    InvalidArgumentException expectedException =
+        new InvalidArgumentException(
+            String.format(
+                "Given classification type %s is not valid according to the configuration.",
+                "UNKNOWN_TYPE"));
```

Comment:
This, on the other hand, has to be adjusted. The InvalidArgumentException doen't have any variables. Thus, its message has to be checked. AssertJ has a way better API than comparing them like this.